contracts/dOrgProjectFactory.sol

```diff
@@ -22,31 +22,134 @@ interface GnosisSafe {
-     */
+    uint256 public projectIndex = 0;
+    mapping(uint256 => Project) public Projects;
+    mapping(uint256 => mapping(address => bool)) public Votes;
+    mapping(uint256 => address[]) public Voters;
+    mapping(uint256 => address[]) public Owners;
+    mapping(uint256 => address) public Finders;
+    mapping(uint256 => uint256) public Thresholds;
+    mapping(uint256 => address) public GnosisSafes;
 
     constructor(address _gnosisLogic) {
         treasuryWallet = address(0x15344EcDc2c4EDFCB092E284d93c20F0529FD8a6);
         dOrgProjectLogic = address(new dOrgProject());
         gnosisLogic = _gnosisLogic;
     }
 
-    function createProject(
-        string calldata projectName,
-        address finderWallet,
-        address[] calldata owners,
+    function newProject(
+        string memory metadataURI,
+        address[] memory owners,
+        address finder,
         uint256 threshold
-    ) external {
+    ) public {
+        Owners[projectIndex] = owners;
+        Finders[projectIndex] = finder;
+        Thresholds[projectIndex] = threshold;
+        Projects[projectIndex] = Project(
+            metadataURI,
+            0,
+            0,
+            block.timestamp,
+            address(0)
+        );
+        projectIndex++;
+    }
+
+    function getProjectIndex() public view returns (uint256){
+        return projectIndex;
+    }
+
+    function getProject(uint256 i) public view returns (Project memory) {
+        return Projects[i];
+    }
+
+    function getProjectGnosisSafe(uint256 i) public view returns (address) {
+        return GnosisSafes[i];
+    }
+
+    function getProjects(uint256 startIndex, uint256 endIndex)
+        public
+        view
+        returns (Project[] memory)
+    {
+        uint256 arraySize = 1 + endIndex - startIndex;
+        require(arraySize > 0, 'Invalid start or end index.');
+        require(endIndex <= projectIndex, 'End index out of range.');
+        Project[] memory projectSlice = new Project[](arraySize);
+        for (uint256 i = startIndex; i <= endIndex; i++) {
+            projectSlice[i] = Projects[i];
+        }
+        return projectSlice;
+    }
 
+    function getVoters(uint256 i) public view returns (address[] memory) {
+        return Voters[i];
+    }
+
+    function getVote(uint256 i, address voter) public view returns (bool) {
+        require(
+            Votes[i][voter],
+            "This address has not voted on the specified proposal."
+        );
+        return Votes[i][voter];
+    }
+
+    function vote(uint256 i, bool approval) public {
+        for(uint v = 0; v<Voters[i].length; v++){
+            require(Voters[i][v] != msg.sender, 'Cannot vote twice.');
+        }
+        require(
+            block.timestamp < Projects[i].createdAt + 604800,
+            "Voting period has closed."
+        );
+        if (approval == true) {
+            Projects[i].forVotes += 1;
+        }
+        if (approval == false) {
```

Comment:
This is a bit nitpicky but this could be just an `else` statement.

Reply:
True. Edited this.